src/client/components/Auth.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { Redirect } from 'react-router-dom';
+
+const Authorization = (allowedRoles: string[]) => (WrappedComponent: React.ComponentClass) => {
+    return class WithAuthorization extends React.Component<{}, { user: {role: string} }> {
+        constructor(props: any) {
+            super(props);
+
+            this.state = {
+                user: {
+                    role: '',
+                }
+            };
+        }
+
+        async componentDidMount() {
+            const self = this;
+            const userModel = ['Pending', 'Judge', 'Admin', 'Owner', 'None'];
+            let result = await fetch('/auth/user_data/class', {
```

Comment:
this can sit for now, but we need to reduxify this for a single source of state. then we won't be making a request for every component rerender.